lib/service/InitService.ts

```diff
@@ -85,6 +86,54 @@ class InitService extends EventEmitter {
 
     return this.swapClientManager.unlockWallets(password);
   }
+
+  public restoreNode = async (args: { password: string, seedMnemonicList: string[] }) => {
+    const { password, seedMnemonicList } = args;
+    if (password.length < 8) {
+      // lnd requires 8+ character passwords, so we must as well
+      throw errors.INVALID_ARGUMENT('password must be at least 8 characters');
+    }
+    if (seedMnemonicList.length !== 24) {
+      throw errors.INVALID_ARGUMENT('mnemonic must be exactly 24 words');
+    }
+    if (this.nodeKeyExists) {
+      throw errors.UNIMPLEMENTED;
+    }
+    if (this.swapClientManager.misconfiguredClientLabels.length > 0) {
+      throw swapErrors.SWAP_CLIENT_MISCONFIGURED(this.swapClientManager.misconfiguredClientLabels);
+    }
+    if (this.pendingCall) {
+      throw errors.PENDING_CALL_CONFLICT;
+    }
```

Comment:
These checks are the same as the ones of the `createNode` method (apart from the the one that checks the length of the mnemonic). Maybe put them in a private method to ensure that the checks and the error messages are consistent between the `create` and `restore` call.

Reply:
I had thought about this but wasn't sure how to do it best given the three methods all make a slightly different set of checks, but I'll give this another pass.